src/Common/GuardedPoolAllocatorOptions.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+namespace clickhouse_gwp_asan
+{
+
+namespace options
+{
+
+struct Options
+{
+    /// Read the options from the included definitions file.
+    /// Use macros to declare variables.
+#define GWP_ASAN_OPTION(type, name, default_value, description) type name = default_value;
+#include <Common/GuardedPoolAllocatorOptions.inc>
+#undef GWP_ASAN_OPTION
+
+    /// Use macros in setDefaults to set default values
+    void setDefaults()
+    {
+#define GWP_ASAN_OPTION(type, name, default_value, description) name = default_value;
+#include <Common/GuardedPoolAllocatorOptions.inc>
+#undef GWP_ASAN_OPTION
+    }
+};
+
+// Parse the options from the provided string.
+bool initOptions(const char * options_str);
```

Comment:
But this way we allow GWP options to be initialized not only by an env variable, but also for providing a string. As of right now it is not used, but I guess it may come in handy later. Do you think I should still remove it?

Reply:
To me it looks like it is better to not export it right now if it not required, but it is not a major thing.